pkg/controllers/cloud/cloudentityselector_controller.go

```diff
@@ -84,9 +129,48 @@ func (r *CloudEntitySelectorReconciler) SetupWithManager(mgr ctrl.Manager) error
 		}); err != nil {
 		return err
 	}
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&cloudv1alpha1.CloudEntitySelector{}).
-		Complete(r)
+
+	if err := ctrl.NewControllerManagedBy(mgr).For(&cloudv1alpha1.CloudEntitySelector{}).Complete(r); err != nil {
+		return err
+	}
+	return mgr.Add(r)
+}
+
+func (r *CloudEntitySelectorReconciler) Start(context.Context) error {
+	if r.initialized {
+		r.Log.Info(controllerDuplicateInit)
+		return nil
+	}
+	// Is cache synced?
+	if !(*r.Mgr).GetCache().WaitForCacheSync(context.TODO()) {
+		return fmt.Errorf("failed to sync shared informer cache")
+	}
+
+	for {
```

Comment:
why its in for loop?? you will retry indefinitely.. fail the controller and exit to inform cannot sync

Reply:
Removed it.